src/draw.c

```diff
@@ -167,24 +167,6 @@ static struct color layout_get_sepcolor(struct colored_layout *cl,
         }
 }
 
-static int get_horizontal_text_icon_padding()
```

Comment:
I think you should keep these functions. `text_icon_padding` may be 0, signalling that normal padding should be used.

Reply:
I see, I may have misinterpreted what you meant. Should I undo https://github.com/dunst-project/dunst/pull/985/commits/4bdf2aba4179e74c7470c5ee5eb639bb5e0c1854 where I removed the functions and changed the height calculations to just use `text_icon_padding`?

Reply:
Yep, that's good. And then you should expand the check to choose between the `text_icon_padding` and horizontal/vertical vertical padding number. This check needs the icon position and if an icon exists. For the horizontal for example that would look something like:
```
static int get_horizontal_text_icon_padding(struct notifiction n)
{
        if (settings.text_icon_padding && n->icon && (n->icon_position == ICON_LEFT || n->icon_position == ICON_RIGHT)) {
                return settings.text_icon_padding;
        } else {
                return settings.h_padding;
        }
}
```

You may also check for this another way, but the functionality should be the same.